cmake/modules/OmrCompilerSupport.cmake

```diff
@@ -329,23 +329,13 @@ function(create_omr_compiler_library)
 	# Add the contents of the macro to CORE_COMPILER_OBJECTS in this scope.
 	# The library name parameter is currently ignored.
 	macro(compiler_library libraryname)
-		# Since CMake *copies* the enviornment from parent to child scope,
-		# a PARENT_SCOPE set ultimately is not reflected for local lookups.
-		# As a result, we need to _also_ update the local scope here if we
-		# want this to succeed when called multiple times in a scope.
-		set(CORE_COMPILER_OBJECTS ${CORE_COMPILER_OBJECTS} ${ARGN}             )
-		set(CORE_COMPILER_OBJECTS ${CORE_COMPILER_OBJECTS} ${ARGN} PARENT_SCOPE)
+		list(APPEND CORE_COMPILER_OBJECTS ${ARGN})
```

Comment:
Can you explain the differences between the lines you removed and the line you added?

Reply:
The second line, `set(CORE_COMPILER_OBJECTS ${CORE_COMPILER_OBJECTS} ${ARGN} PARENT_SCOPE)` is no longer required since we are no longer introducing a new scope. 
`set(CORE_COMPILER_OBJECTS ${CORE_COMPILER_OBJECTS} ${ARGN})`, is effectively the same as the simple `list(APPEND ...)` although I feel the `list(APPEND...)` form more clearly shows our intent

Reply:
@rwy0717 I think this is good.  Your thoughts?